### ISSUE TITLE
Ignored Scope Config ends in StoreNotFound Issue during Setup Upgrade

### ISSUE THREAD

**Issue by DavidLambauer** [mollie/magento2#287]
Hi, we're using version 1.16.0.x-dev. When we're running a setup:upgrade, we receive the following issue 

``` 
Module 'Mollie_Payment':
Upgrading data.. The store that was requested wasn't found. Verify the store and try again.
```

I digged into it a little and found out that `\Mollie\Payment\Setup\UpgradeData::updateCustomerReturnUrl` iterates through websites and calls `\Mollie\Payment\Setup\UpgradeData::updateCustomerReturnUrlForScope`. In this method, you try to get the value of `payment/mollie_general/custom_redirect_url` which might be set in the scope of a website. The issue is, that the scope is not forwarded to `\Mollie\Payment\Config::customRedirectUrl` and therefore the config of a store is called. In our case, there is a website with id 2 but not a store which finally causes the exception described above. 

The potential solution would be to forward the scope to the config resolver, so that the values of the website can be received.

Our Magento Version btw is 2.3.5-p2

**Comment by michielgerritsen** [comment]
Hi David,

Thanks for pointing this out. I've just created PR #288. Can you check if this fixes the issue for you?

**Comment by DavidLambauer** [comment]
Hi @michielgerritsen , I think in the `\Mollie\Payment\Setup\UpgradeData::updateCustomerReturnUrlForScope` is still an issue. 

The scope is missing:

```php
$value = $this->mollieConfig->customRedirectUrl($scopeId);
```

I think, this is better:

```php
$value = $this->mollieConfig->customRedirectUrl($scopeId, $scope);
```

**Comment by AlexNauta** [comment]
We also get Upgrading data.. The store that was requested wasn't found. Verify the store and try again.

Also after applying the fixes from PR #288 ánd DavidLambauer's tip.

Could it be, because we have no store 1? (that was a problem with multiple other modules in the past)

**Comment by michielgerritsen** [comment]
Hi @DavidLambauer and @AlexNauta,

The PR is updated with a new approach. Can you check if this solves your problem?

**Comment by DavidLambauer** [comment]
Hi @michielgerritsen looks good to me!

**Comment by Frank-Magmodules** [comment]
Hi @DavidLambauer , thank you again for the detailed issue report and thinking with us! We are happy to share that we've just released the new [1.16.1](https://github.com/mollie/magento2/releases/tag/v1.16.1) version with a fix for this issue. We are closing this issue now but please feel free to reopen the issue if this still occurs.
